analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java

```diff
@@ -0,0 +1,88 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015,2016 
+*/
+package quarks.analytics.sensors;
+
+import java.util.concurrent.TimeUnit;
+
+import quarks.function.Function;
+import quarks.function.Predicate;
+import quarks.topology.TStream;
+
+/**
+ * Filters aimed at sensors.
+ */
+public class Filters {
+
+    /**
+     * Deadband filter with maximum suppression time.
+     * 
+     * A filter that discards any tuples that are in the deadband, uninteresting to downstream consumers.
+     * <P>
+     * A tuple {@code t} is passed through the deadband filter if:
+     * <UL>
+     * <LI>
+     * {@code deadband.test(value.apply(t))} is false, that is the tuple's value is outside of the deadband
+     * </LI>
+     * <LI>
+     * OR {@code deadband.test(value.apply(t))} is true AND the last tuple's value was outside of the deadband.
+     * This corresponds to the first tuple's value inside the deadband after a period being outside it.
+     * </LI>
+     * <LI>
+     * OR it has been more than {@code maximumSuppression} seconds (in unit {@code unit}) 
+     * </LI>
+     * <LI>
+     * OR it is the first tuple (effectively the state of the filter starts as outside of the deadband).
+     * </LI>
+     * </UL>
+     * 
+     * @param <T> Tuple type.
+     * @param <V> Value type for the deadband function.
+     * 
+     * @param stream Stream containing readings.
+     * @param value Function to obtain the tuple's value passed to the deadband function.
+     * @param deadband Function that defines the deadband.
+     * @param maximumSuppression Maximum amount of time to suppress values that are in the deadband.
+     * @param unit Unit for {@code maximumSuppression}.
+     * @return Filtered stream.
+     */
+    public static <T, V> TStream<T> deadband(TStream<T> stream, Function<T, V> value, Predicate<V> deadband,
```

Comment:
Might just be me but I'm finding things confusing because the name "deadband" is used multiple times but with opposite semantics.  

i.e., ``Predicate deadband`` is testing if the value is IN the deadband - i.e., uninteresting values, whereas ``static deadband()``, something with the exact same name, is selecting tuples OUT of the deadband - i.e., interesting values.

``DeadbandFilter`` also has the OUT-of-deadband semantics and also takes the ``Predicate deadband`` which has the IN-deadband semantics.   Also, I think its not helping that ``TStream.filter()`` doesn't change the sense of what its predicate selects, "filter()" doesn't "filter out" what the predicate selects, but ``DeadbandFilter`` does.

Perhaps changing to ``Predicate inDeadband`` will at least help differentiate it & clarify what its semantics are?